### PR TITLE
Allow escalation of memory for FilterConsensusReads

### LINKED ISSUE #57: Hardcoded 8GB limit in the FilterConsensusReads processes
### Description of the bug

There is a hardcoded limit of 8GB in the FilterConsensusReads processes defined in `modules.config`. Aside being on the low side for very deep sequenced datasets (think ctDNA), it relies on retries to increase memory requirements should the task fail. However, what happens with many schedulers like SLURM is that if the task goes out of memory (and with 10,000X data it is a likely possibility) it is killed, causing an error and the premature end of the pipeline (so it won't ever be retried).

I would think this should belong to the user's definitions and should not be hardcoded. 

### Command used and terminal output

_No response_

### Relevant files

_No response_

### System information

- Nextflow: 23.04.2
- Hardware: HPC
- Executor: slrum
- Container Engine: apptainer
- OS: Debian
- nf-core/fastquorum version: 1.0.0

### PR DESCRIPTION
**Pull request by SPPearce**
Fixed #57

### CODE REVIEW

**Comment by SPPearce** [comment]
@nh13 , this is what I meant, it will retry with escalating resource requirements due to the `* task.attempt`

File: conf/base.config [hunk 1]
```diff
@@ -67,4 +62,7 @@ process {
         cpus   = { check_max( 4 * task.attempt, 'cpus' ) }
         memory = { check_max( 12.GB * task.attempt, 'memory' ) }
     }
+    withName:'FILTERCONSENSUSREADS' {
+        memory = { check_max( 12.GB * task.attempt, 'memory' ) }
```
Comment (nh13) [inline conf/base.config:66]: Gotcha.  Why not add the `error_retry` on the process definition as well?

Comment (SPPearce) [reply conf/base.config:66]: A few lines above there is:
```
    errorStrategy = { task.exitStatus in ((130..145) + 104) ? 'retry' : 'finish' }
    maxRetries    = 1
```
Which sets by default retrying for all processes.

Comment (nh13) [reply conf/base.config:66]: I asked one of the Seqera nf-core folks at ISMB where _module_-level config should live and they said not in `base.config` but in `modules.config`.  That's also my reading in the nf-core docs, as well as in other nf-core pipelines (e.g. `withName` are in the `modules.config` or `modules/<process>.config`, whereas `withLabel` are in the `base.config`). The [nf-core docs even say to copy the `modules.config`](https://nf-co.re/docs/usage/getting_started/configuration#understanding-and-modifying-tool-arguments). 

Can you please move it back to `modules.config`?

**Review by nh13 — CHANGES_REQUESTED** [review]
One request to move the `withName` back to `modules.config`, and one conflict with `dev` to resolve.   Thank-you!

**Review by nh13 — APPROVED** [review]
Thank you for doing all this!

we may need a release soon. I’ll take a shot at creating a milestone with issues I think need solved before a release

**Pull request merged**